clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -768,13 +768,14 @@ public FetchManagerMetrics(Metrics metrics, String metricGrpPrefix) {
         }
 
         public void recordTopicFetchMetrics(String topic, int bytes, int records) {
-            Map<String, String> metricTags = new HashMap<>();
-            metricTags.put("topic", topic.replace(".", "_"));
-
             // record bytes fetched
-            String name = "topic." + topic + ".bytes-fetched";
+            String name = String.format("topic.%s.bytes-fetched", topic);
```

Comment:
This should be slower than what you replaced. Do you have any evidence that it's faster?